crates/string_wizard/src/magic_string/replace.rs

```diff
@@ -34,22 +34,16 @@ impl<'text> MagicString<'text> {
     options: ReplaceOptions,
   ) -> &mut Self {
     let to: CowStr<'text> = to.into();
-    // PERF(hyf0): Unnecessary `collect` due to borrow checker limitation.
-    let matches = self
-      .source
-      .match_indices(from)
-      .take(options.count)
-      .map(|(match_start, part)| (match_start, match_start + part.len()))
-      .collect::<Box<[_]>>();
-
-    matches.iter().for_each(|(start, end)| {
+    let source = self.source.clone();
```

Comment:
I can't quite remember why I make

```
    to: impl Into<CowStr<'text>>,
```

instead of just

```
    to: &str,
```

Reply:
Could you try refactor it to use `&str`, I think it should solves the borrow checker problem.

If `to` is `String`, `.clone` does do a heal allocation here.